crawler/lookingglass_crawler/resilient_proxy_middleware.py

```
@@ -0,0 +1,82 @@
+        """
+        If proxy-related exception occurs, retry without proxy.
+        """
+        if 'proxy' in request.meta:
+            logger.warning(f"Proxy failed: {exception}. Retrying without proxy.")
+            # Remove proxy and retry
+            request.meta.pop('proxy', None)
+            return request.replace(dont_filter=True)
+        
+        return None
+    
+    def refresh_proxy_list(self):
```

Ruff (0.11.9)

Missing return type annotation for public function `refresh_proxy_list`

Add return type annotation: `None`

(ANN201)